ert_shared/ensemble_evaluator/evaluator.py

```diff
@@ -234,20 +235,31 @@ def stop(self):
         self._ws_thread.join()
 
     def run_and_get_successful_realizations(self):
-        try:
-            with self.run() as mon:
-                for _ in mon.track():
-                    pass
-        except ConnectionRefusedError as e:
-            logger.debug(
-                f"run_and_get_successful_realizations caught {e}, cancelling or stopping ensemble..."
-            )
-            if self._ensemble.is_cancellable():
-                self._ensemble.cancel()
-            else:
-                self._stop()
-            self._ws_thread.join()
-            raise
+        monitor_context = self.run()
+        unsuccessfull_connection_attempts = 0
+        while True:
+            try:
+                with monitor_context as mon:
+                    for t in mon.track():
+                        unsuccessfull_connection_attempts = 0
+                        print(t)
+                break
+            except ConnectionClosedError as e:
+                logger.debug(
+                    f"Connection closed unexpectedly in run_and_get_successful_realizations: {e}"
+                )
+            except ConnectionRefusedError as e:
+                logger.debug(
+                    f"run_and_get_successful_realizations caught {e}, cancelling or stopping ensemble..."
+                )
+                unsuccessfull_connection_attempts += 1
+                if unsuccessfull_connection_attempts == 3:
```

Comment:
Perhaps the number 3 could be coded as e.g. `MAX_UNSUCCESSFUL_CONNECTION_ATTEMPTS=3` in the module or even locally in the function?